packages/shared/src/clients/rpc/RpcClient2.ts

```diff
@@ -39,6 +45,24 @@ export class RpcClient2 extends ClientCore implements BlockClient {
     return { ...block.data.result }
   }
 
+  async getBalance(holder: EthereumAddress, blockNumber: number | 'latest') {
+    const method = 'eth_getBalance'
+    const encodedNumber =
+      blockNumber === 'latest' ? 'latest' : Quantity.encode(BigInt(blockNumber))
+    const balanceResponse = await this.query(method, [
+      holder.toString(),
+      encodedNumber,
+    ])
+
+    const balance = EVMBalanceResponse.safeParse(balanceResponse)
+    if (!balance.success) {
+      throw new Error(
```

Comment:
please also add here before throwing: logger.warn with strigified balanceResponse, will be used for debugging